### PR TITLE
TDL-16323: Add request timeout

### PR DESCRIPTION
**Pull request by harshpatel4crest**
# Description of change
TDL-16323: Add request timeout
- Use default timeout of 300 seconds if the user does not enter the preferred timeout value

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

**Review by umangagrawal-crest — CHANGES_REQUESTED** [review]
@harshpatel4crest Please update the readme and sample config file

**Comment by harshpatel4crest** [comment]
> @harshpatel4crest Please update the readme and sample config file

Updated readme file and sample config file.

File: tap_klaviyo/utils.py [hunk 1]
```diff
@@ -185,3 +191,17 @@ def transfrom_and_write_records(events, stream):
...
                     event, event_schema, event_mdata
             ))
+
+# return the 'timeout'
+def get_request_timeout():
+    args = singer.utils.parse_args([])
+    # get the value of request timeout from config
+    config_request_timeout = args.config.get('request_timeout')
+
+    if config_request_timeout and float(config_request_timeout):
```
Comment (savan-chovatiya) [inline tap_klaviyo/utils.py:201]: Add detailed comment over if condition.

File: tap_klaviyo/utils.py [hunk 2]
```diff
@@ -185,3 +191,17 @@ def transfrom_and_write_records(events, stream):
...
+        request_timeout = float(config_request_timeout)
+        # return the timeout from config
+        return request_timeout
```
Comment (savan-chovatiya) [inline tap_klaviyo/utils.py:204]:
```suggestion
        return float(config_request_timeout)
```

File: tests/unittests/test_timeout.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        # mock parse args
+        mocked_parse_args.return_value = get_args(mock_config)
+
+        try:
+            # function call
+            utils.authed_get("test_source", "", "")
+        except requests.Timeout:
+            pass
+
+        # verfiy that we backoff for 5 times
```
Comment (savan-chovatiya) [inline tests/unittests/test_timeout.py:137]:
```suggestion
        # verify that we backoff for 5 times
```

File: tests/unittests/test_timeout.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        # mock parse args
+        mocked_parse_args.return_value = get_args(mock_config)
+
+        try:
+            # function call
+            utils.authed_get("test_source", "", "")
+        except requests.ConnectionError:
+            pass
+
+        # verfiy that we backoff for 5 times
```
Comment (savan-chovatiya) [inline tests/unittests/test_timeout.py:155]:
```suggestion
        # verify that we backoff for 5 times
```

File: tap_klaviyo/utils.py (hunk 1 above)
Comment (dbshah1212) [reply tap_klaviyo/utils.py:201]: I can't see the meaning of using the request_timeout variable.

File: tests/unittests/test_timeout.py (hunk 4 above)
Comment (harshpatel4crest) [reply tests/unittests/test_timeout.py]: Updated.

File: tests/unittests/test_timeout.py (hunk 3 above)
Comment (harshpatel4crest) [reply tests/unittests/test_timeout.py]: Updated.

File: tap_klaviyo/utils.py (hunk 2 above)
Comment (harshpatel4crest) [reply tap_klaviyo/utils.py]: Updated.

File: tap_klaviyo/utils.py (hunk 1 above)
Comment (harshpatel4crest) [reply tap_klaviyo/utils.py:202]: Updated the code and returned the value directly.

**Review by savan-chovatiya — APPROVED** [review]

**Review by umangagrawal-crest — APPROVED** [review]

**Review by namrata270998 — APPROVED** [review]

**Review by namrata270998 — APPROVED** [review]

**Review by karanpanchal-crest — APPROVED** [review]

**Review by KrisPersonal — APPROVED** [review]

**Review by kspeer825 — APPROVED** [review]

**Pull request merged**